ksp/build.gradle.kts

```diff
@@ -5,19 +5,45 @@ plugins {
   alias(libs.plugins.mavenPublish)
 }
 
-tasks.withType<KotlinCompile>()
+tasks
+  .withType<KotlinCompile>()
   .matching { it.name.contains("test", ignoreCase = true) }
   .configureEach {
-    compilerOptions {
-      freeCompilerArgs.add("-opt-in=org.jetbrains.kotlin.compiler.plugin.ExperimentalCompilerApi")
-    }
+    compilerOptions { optIn.add("org.jetbrains.kotlin.compiler.plugin.ExperimentalCompilerApi") }
   }
 
+// It's not possible to test both KSP 1 and KSP 2 in the same compilation unit
+val testKsp2 = providers.systemProperty("kct.test.useKsp2").getOrElse("false").toBoolean()
+
+tasks.test {
+  if (testKsp2) {
+    systemProperty("kct.test.useKsp2", testKsp2)
+  }
+}
+
 dependencies {
   compileOnly(libs.kotlin.compilerEmbeddable)
   api(projects.core)
+
+  // KSP's AA dependencies point to the wrong common-deps artifact
+  compileOnly(libs.ksp.aa) { exclude(group = "com.google.devtools.ksp", module = "common-deps") }
```

Comment:
can we try to replace it with `libs.ksp.aa.embeddable`?

Reply:
Was able to clean up some in [c9490a4](https://github.com/ZacSweers/kotlin-compile-testing/pull/196/commits/c9490a452e36466f60317aa926e9fcb4fc12ca9b)